### PR TITLE
Yarn 4 + ESLint 9 + PnP Support (3)

### PR DESCRIPTION
**Pull request by SleeplessByte**
It keeps breaking on GH (probably the diff) so just recreating as that seems to trigger a working CI once.

- **Upgrade to Yarn 4, ESLint 9, and add PnP support**
- **Upgrade SDKs**
- **Sync**
- **Format files**
- **Fix .vscode files, upgrade all tests to new format**
- **Update workflows to Node LTS @ 20**
- **Fix lint issues**
- **Remove babel transpilation for common actions**
- **Simplify changed files check**

### CODE REVIEW

**Comment by SleeplessByte** [comment]
(Please do not merge yet, test-runner needs to be updated first)

File: .prettierrc [hunk 1]
```diff
@@ -1,15 +1,10 @@
+
```
Comment (ErikSchierboom) [inline .prettierrc:1]: Did you intend to have this empty line?

**Review by ErikSchierboom — APPROVED** [review]

**Comment by ErikSchierboom** [comment]
I'm sure you noticed, but please merge this with `[no important files changed]` in the commit message.

File: .prettierrc (hunk 1 above)
Comment (SleeplessByte) [reply .prettierrc:1]: Not necessarily. But it doesn't do anything :P

**Comment by SleeplessByte** [comment]
@ErikSchierboom no let's rerun the tests on EVERY SINGLE SOLUTION :') /s

Technically `[no important files changed]` is even correct as none of the tests or content has changed.

**Pull request merged**